### PR TITLE
Add copy image support.

### PR DESCRIPTION
**Pull request by keltar**
Only works with gboard virtual keyboard.

### CODE REVIEW

File: app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java [hunk 1]
```diff
@@ -33,15 +33,11 @@ protected void onCreate(Bundle savedInstanceState) {
         Intent intent = getIntent();
         String action = intent.getAction();
         String type = intent.getType();
+
         if (!Intent.ACTION_SEND.equals(action) || type == null) {
             finish();
             return;
         }
-        if (!type.startsWith("text/")) {
```
Comment (talhacohen) [inline app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java:40]: Can we extract the switch case to a method and change this to:

if (type.startsWith("text/")) {
   ...
} else if type.startsWith("text/")) {
   handleSendImage(intent);
} else {
  handleSendText(intent, R.string.error_type_not_supported_by_platform);
}

File: app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java [hunk 2]
```diff
@@ -114,6 +116,23 @@ private void appendVCardTypes(StringBuilder output, List <? extends VCardParamet
         }
     }
 
+    private void handleSendImage(Intent intent) {
+        boolean has_extra_steam = intent.hasExtra(Intent.EXTRA_STREAM);
+        if(!has_extra_steam) {
```
Comment (talhacohen) [inline app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java:121]: Missing space after the if
Should be `if (!has_extra_steam)`

File: app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java [hunk 3]
```diff
@@ -114,6 +116,23 @@ private void appendVCardTypes(StringBuilder output, List <? extends VCardParamet
...
+            return;
```
Comment (talhacohen) [inline app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java:122]: Can we add an error toast here? Can this happen for users?

File: app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java [hunk 4]
```diff
@@ -114,6 +116,23 @@ private void appendVCardTypes(StringBuilder output, List <? extends VCardParamet
...
+        Uri uri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+        ClipboardManager clipboard = (android.content.ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+        ClipData clip = ClipData.newUri(getContentResolver(), "image", uri);
+        clipboard.setPrimaryClip(clip);
+
+        if (PreferenceUtil.shouldDisplayNotification(this)) {
+            NotificationUtil.createNotification(this);
+        } else {
+            showToast(getString(R.string.notification_title));
+        }
```
Comment (talhacohen) [inline app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java:133]: Can you extract this to a method and call it here and in copyToClipboard?

File: app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java [hunk 5]
```diff
@@ -114,6 +116,23 @@ private void appendVCardTypes(StringBuilder output, List <? extends VCardParamet
...
+    private void handleSendImage(Intent intent) {
+        boolean has_extra_steam = intent.hasExtra(Intent.EXTRA_STREAM);
+        if(!has_extra_steam) {
+            return;
+        }
+        Uri uri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+        ClipboardManager clipboard = (android.content.ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+        ClipData clip = ClipData.newUri(getContentResolver(), "image", uri);
+        clipboard.setPrimaryClip(clip);
+
```
[the remark this answers is not in the document]
Comment (keltar) [reply app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java]: Copying kind of works (i.e. not produces an error) even with old android versions (tested on 4.4), but it only gives virtual filename for shared image, and there is no gboard for 4.4 so can't test pasting anywhere. I think apps could accept image pasting starting with android 7.1. Chrome-like browsers have "copy image" for about two years now, and many chat or note-taking apps support pasting, but e.g. gallery or firefox don't have image copying, so share_to_clipboard could be handy.

[the remark this answers is not in the document]
Comment (talhacohen) [reply app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java]: In that case, can we add a check that the Android version is at least 7.1? That should be a straightforward change and will not mislead the users with old devices. Though I doubt we have a lot of users with such an old device :)

File: app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java [hunk 6]
```diff
@@ -33,25 +33,22 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            default:
-                handleSendText(intent, R.string.error_type_not_supported);
-                break;
+
+        if (type.equals("text/plain")) {
+            handleSendText(intent, R.string.error_no_data);
+        } else if (type.equals("text/x-vcard")) {
+            handleSendVCard(intent);
+        } else if (type.startsWith("image/")) {
+            // copy shared image to clipboard
```
Comment (talhacohen) [inline app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java:47]: Can probably remove the comment. it's clear from the method

**Comment by talhacohen** [comment]
Generally looks good to me, lets add a check for the android version.

Thank you very much for doing this!

**Pull request merged**